server/form-pages/assess/matchingInformation/matchingInformationTask/matchingInformation.ts

```diff
@@ -182,8 +182,10 @@ export default class MatchingInformation implements TasklistPage {
       errors.lengthOfStayAgreed = 'You must state if you agree with the length of the stay'
     }
 
-    if (this.body.lengthOfStayAgreed === 'no' && !this.body.lengthOfStayWeeks && !this.body.lengthOfStayDays) {
-      errors.lengthOfStay = 'You must provide a recommended length of stay'
+    if (this.body.lengthOfStayAgreed === 'no') {
+      if (!isCardinal(this.body.lengthOfStayWeeks) || !isCardinal(this.body.lengthOfStayDays)) {
```

Comment:
I’m not sure about requiring that _both_ fields are filled in. A common use case seems to be suggesting an 8-week placement instead of a 12-week one: in this case it seems like unnecessary friction to force the user to enter 0 for days.

Given we work out the resulting length in days from both fields, couldn’t the validation simply ensure the resulting length is more than `0` days, from any combination of weeks and/or days? The error message would still work fine. The `isCardinal` validation would still be useful, but only if any value has been entered (with the above suggestion to ignore whitespace).

Reply:
Yes, I was thinking that.  This ia  bit belt'n'braces as this level of validation allows the existing downstream handling to be left untouched.
I'm now thinking it would be better to fix the handling so this field validation could be relaxed.